src/vtquery.cpp

```diff
@@ -163,12 +190,97 @@ struct CompareDistance {
     }
 };
 
+std::vector<vtzero::index_value_pair> get_comparison_tags(vtzero::feature f) {
+    std::vector<vtzero::index_value_pair> v;
+    while (auto ii = f.next_property_indexes()) {
+        v.push_back(ii);
+    }
+    return v;
+}
+
+// compares a vector of property tags to determine deduplication of features
+bool compare_comparison_tags(std::vector<vtzero::index_value_pair> a, std::vector<vtzero::index_value_pair> b) {
```

Comment:
These `std::vector<vtzero::index_value_pair>` args should be passed as `std::vector<vtzero::index_value_pair> const&` to avoid expensive copies.